workload/metrics.go

```diff
@@ -9,26 +12,39 @@ var (
 			Name: "operations_total",
 			Help: "How many user operations are attempted, partitioned by operation.",
 		},
-		[]string{"operation"},
+		[]string{"operation", "state"},
 	)
 	opsFailed = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Name: "operations_failed_total",
 			Help: "How many user operations failed, partitioned by operation.",
 		},
-		[]string{"operation"},
+		[]string{"operation", "state"},
 	)
 	opDuration = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Name:    "operation_duration_milliseconds",
 			Help:    "Duration of user operations in milliseconds, partitioned by operation.",
 			Buckets: []float64{25.0, 50.0, 100.0, 250.0, 500.0, 1500.0, 2500.0},
 		},
-		[]string{"operation"},
+		[]string{"operation", "state"},
 	)
 
 	// Maps from the operation to an attempted/failed metric labelled with the operation
-	attemptMetrics  = map[string]prometheus.Counter{}
-	failedMetrics   = map[string]prometheus.Counter{}
-	durationMetrics = map[string]prometheus.Observer{}
+	attemptMetrics  = map[string]map[string]prometheus.Counter{}
+	failedMetrics   = map[string]map[string]prometheus.Counter{}
+	durationMetrics = map[string]map[string]prometheus.Observer{}
+
+	States   = []string{"warmup", "steady", "warm_down"}
```

Comment:
rampdown might be a better choice of words?  but then the former should be rampup?  I think tpc-c uses ramp-up and ramp-down.  Maybe warmup and cooldown?